test/test-spawn.c

```diff
@@ -1079,6 +1083,60 @@ TEST_IMPL(argument_escaping) {
   return 0;
 }
 
+TEST_IMPL(spawn_batch_args) {
+  int r, i;
+  char* test_args[] = {
```

Comment:
I think a good test would actually *spawn* a .batch file and verifies that the arguments come through correctly. The reason is that nobody really understand cmd's command line parsing.
A way to do this is by having the batch file print all arguments to stdout on a separate line, and verifies that the output is what you expected.

```
@echo off
:loop
if "%1"=="" goto :eof
echo %1
shift
goto loop
```